### PR TITLE
Remove dependency from aws-sdk-promise. r=jhford

### PR DESCRIPTION
**Pull request by walac**
Amazon has added the .promise() function to aws-sdk and aws-sdk-promise
isn't required anymore. Furthermore, aws-sdk-promise is going to be
deprecated. Details: http://tinyurl.com/hry2fa3

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,6 @@
   "homepage": "https://github.com/taskcluster/docker-worker",
   "dependencies": {
     "aws-sdk": "^2.1.5",
-    "aws-sdk-promise": "0.0.2",
```
Comment (gregarndt) [inline package.json]: might need to bump the aws-sdk libarry and make sure it works.

**Comment by jhford** [comment]
I don't know the docker worker codebase very well, but this change looks like the right thing to do.  Also, looking at the output of `git grep "\.data"`, I don't see anything where we use the now-removed `.data` parameter

r+ from me, with the caveat that I don't know the codebase well.

**Comment by walac** [comment]
@gregarndt want to give a last review before merge?

**Comment by gregarndt** [comment]
as far as I'm aware the aws-sdk is only used directly by that upload schema file, so I think this should be ok.

**Pull request merged**